src/app/settings/organization-plans.component.ts

```diff
@@ -100,6 +64,137 @@ export class OrganizationPlansComponent {
         this.selfHosted = platformUtilsService.isSelfHost();
     }
 
+    async ngOnInit() {
+        const plans = await this.apiService.getPlans();
+        this.plans = plans.data;
+        this.loading = false;
+    }
+
+    get createOrganization() {
+        return this.organizationId == null;
+    }
+
+    get productTypes() {
+        return ProductType;
+    }
+
+    get selectedPlan() {
+        return this.plans.find((plan) => plan.type === this.plan);
+    }
+
+    get selectableProducts() {
+        let validPlans = this.plans;
+
+        if (this.ownedBusiness) {
+            validPlans = validPlans.filter((plan) => plan.canBeUsedByBusiness);
+        }
+
+        if (!this.showFree) {
+            validPlans = validPlans.filter((plan) => plan.product !== ProductType.Free);
+        }
+
+        validPlans = validPlans
+            .filter((plan) => !plan.legacyYear
+                && !plan.disabled
+                && (plan.isAnnual || plan.product === this.productTypes.Free));
+
+        return validPlans;
+    }
+
+    get selectablePlans() {
+        return this.plans.filter((plan) => !plan.legacyYear && !plan.disabled && plan.product === this.product);
+    }
+
+    additionalStoragePriceMonthly(selectedPlan: PlanResponse) {
+        if (!selectedPlan.isAnnual) {
+            return selectedPlan.additionalStoragePricePerGb;
+        }
+        return selectedPlan.additionalStoragePricePerGb / 12;
+    }
+
+    seatPriceMonthly(selectedPlan: PlanResponse) {
+        if (!selectedPlan.isAnnual) {
+            return selectedPlan.seatPrice;
+        }
+        return selectedPlan.seatPrice / 12;
+    }
+
+    additionalStorageTotal(plan: PlanResponse): number {
+        if (!plan.hasAdditionalStorageOption) {
+            return 0;
+        }
+
+        return plan.additionalStoragePricePerGb * Math.abs(this.additionalStorage || 0);
+    }
+
+    seatTotal(plan: PlanResponse): number {
+        if (!plan.hasAdditionalSeatsOption) {
+            return 0;
+        }
+
+        return plan.seatPrice * Math.abs(this.additionalSeats || 0);
+    }
+
+    get subtotal() {
+        let total = this.selectedPlan.basePrice;
+        if (this.selectedPlan.hasAdditionalSeatsOption && this.additionalSeats) {
+            total += this.seatTotal(this.selectedPlan);
+        }
+        if (this.selectedPlan.hasAdditionalStorageOption && this.additionalStorage) {
+            total += this.additionalStorageTotal(this.selectedPlan);
+        }
+        if (this.selectedPlan.hasPremiumAccessOption && this.premiumAccessAddon) {
+            total += this.selectedPlan.premiumAccessOptionPrice;
+        }
+        return total;
+    }
+
+    // TODO: Wire up to chosen API
+    get salesTax() {
+        return 0;
```

Comment:
Missed this before... we actually want to use our our API for calculating sales tax... then any changes really happen server-side for us, not here in the UI. It would be our API that would return `0` for the time being (this would actually call the API with the necessary parameters to calculate sales tax, and then only for US).

Reply:
Removed